data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -177,17 +178,21 @@ static DataServiceException createExceptionFromResponse(String url, HttpResponse
             boolean isPermanent = false;
             if (!StringUtils.isBlank(errorFromResponse)) {
                 try {
-                    JSONObject jsonObject = new JSONObject(errorFromResponse);
+                    JsonNode jsonObject = new ObjectMapper().readTree(errorFromResponse);
                     if (jsonObject.has("error")) {
                         formattedException = new DataWebException(errorFromResponse, httpResponse, thrownException);
                         OneApiError apiError = ((DataWebException) formattedException).getApiError();
                         message = apiError.getDescription();
                         isPermanent = apiError.isPermanent();
                     } else if (jsonObject.has("message")) {
-                        message = jsonObject.getString("message");
+                        message = jsonObject.get("message").asText();
                     }
-                } catch (JSONException ex) {
+                } catch (JsonMappingException e) {
                     // It's not ideal to use an exception here for control flow, but we can't know if it's a valid JSON until we try to parse it
+                    LOGGER.error("json mapping error happened while parsing errorFromResponse" + e.getMessage(), e);
+                } catch (JsonProcessingException e) {
+                    // It's not ideal to use an exception here for control flow, but we can't know if it's a valid JSON until we try to parse it
+                    LOGGER.error("json processing error happened while parsing errorFromResponse" + e.getMessage(), e);
```

Comment:
Nitpick , LOGGER  or log as the variable name. We have it mixed up. Lets use placeholder markers {0} instead of concatenation